connectors/src/connectors/slack/chat/citations.ts

```diff
@@ -18,12 +25,24 @@ export function annotateCitations(
   content: string,
   action: AgentActionType | null
 ): { formattedContent: string; footnotes: SlackMessageFootnotes } {
-  const references: { [key: string]: RetrievalDocumentType } = {};
+  const references: {
+    [key: string]: RetrievalDocumentType | WebsearchResultType;
+  } = {};
 
   if (action && isRetrievalActionType(action) && action.documents) {
```

Comment:
You could avoid adding extra typeguards by building the slack citation in this loop.